zookeeper-docs/src/main/resources/markdown/zookeeperTools.md

```diff
@@ -205,6 +206,86 @@ USAGE: SnapshotFormatter [-d|-json] snapshot_file
+
+The delta includes specific znode paths added, updated, deleted comparing one snapshot to another. 
+
+It's useful in use cases that involve snapshot analysis, such as offline data consistency checking, and data trending analysis (e.g. what's growing under which zNode path during when).
+
+This tool only outputs information about permanent nodes, ignoring both sessions and ephemeral nodes.
+
+It provides two tuning parameters to help filter out noise: 
+1. `--nodes` Threshold number of children added/removed; 
+2. `--bytes` Threshold number of bytes added/removed.
+
+#### Locate Snapshots
+Snapshots can be found in [Zookeeper Data Directory](zookeeperAdmin.html#The+Data+Directory) which configured in [conf/zoo.cfg](zookeeperStarted.html#sc_InstallingSingleMode) when set up Zookeeper server. 
+
+#### Supported Snapshot Formats
+This tool supports uncompressed snapshot format, and compressed snapshot file formats: `snappy` and `gz`. Snapshots with different formats can be compared using this tool directly without decompression.
+
+#### Running the Tool
+Running the tool with no command line argument or an unrecognized argument, it outputs the following help page:
+
+```
+usage: java -cp <classPath> org.apache.zookeeper.server.SnapshotComparer
+ -b,--bytes <BYTETHRESHOLD>       (Required) The node data delta size threshold, in bytes, for printing the node.
+ -d,--debug <DEBUG>               Use debug output.
+ -i,--interactive <INTERACTIVE>   Enter interactive mode.
+ -l,--left <LEFT>                 (Required) The left snapshot file.
+ -n,--nodes <NODETHRESHOLD>       (Required) The descendant node delta size threshold, in nodes, for printing the node.
+ -r,--right <RIGHT>               (Required) The right snapshot file.
+```
+Example Command:
+
+```
+$ bin/zkSnapshotComparer.sh -l /zookeeper-data/backup/snapshot.0.snappy -r /zookeeper-data/backup/snapshot.8.gz -b 100000 -n 100
+```
+
+Example Output:
+
+```
+...
+Deserialized snapshot in snapshot.0.snappy in 16.496096 seconds
+Processed data tree in 6.332329 seconds
+2020-01-20 17:14:08,940 [myid:] - INFO  [main:WatchManagerFactory@42] - Using org.apache.zookeeper.server.watch.WatchManager as watch manager
+2020-01-20 17:14:08,940 [myid:] - INFO  [main:WatchManagerFactory@42] - Using org.apache.zookeeper.server.watch.WatchManager as watch manager
+Deserialized snapshot in snapshot.8.gz in 0.001641 seconds
+Processed data tree in 0.000118 seconds
+Node count: 8129
+Total size: 121341
+Max depth: 6
+Count of nodes at depth 1: 1
+Count of nodes at depth 2: 12
+Count of nodes at depth 3: 17
+Count of nodes at depth 4: 21
+Count of nodes at depth 5: 720
+Count of nodes at depth 6: 7358
+
+Node count: 8
+Total size: 101000 
+Max depth: 4
+Count of nodes at depth 1: 1
+Count of nodes at depth 2: 2
+Count of nodes at depth 3: 4
+Count of nodes at depth 4: 1
+
+Analysis for depth 0
+Node  found in both trees. Delta: -20341 bytes, -8129 descendants
```

Comment:
Can you please add more docs about how to use the `interactive` mode?

Reply:
good point - I added an `interactive` mode session in the docs.  I also made some changes to make the tool more user-friendly in general, especially under interactive mode.